src/app/tests/TestDefaultTermsAndConditionsProvider.cpp

```diff
@@ -0,0 +1,229 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#define CHIP_CONFIG_TC_REQUIRED 1                          // Enable conditional feature for testing
+#define CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS 0         // Dummy define to satisfy #error check
+#define CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION 0 // Dummy define to satisfy #error check
+
+#include "app/server/DefaultTermsAndConditionsProvider.h"
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/TestPersistentStorageDelegate.h>
+#include <pw_unit_test/framework.h>
+
+TEST(DefaultTermsAndConditionsProvider, TestInitSuccess)
+{
+    CHIP_ERROR err;
+
+    chip::TestPersistentStorageDelegate storageDelegate;
+    chip::app::DefaultTermsAndConditionsProvider tncProvider;
+
+    uint16_t requiredAcknowledgements        = 1;
+    uint16_t requiredAcknowledgementsVersion = 1;
+    err = tncProvider.Init(&storageDelegate, requiredAcknowledgements, requiredAcknowledgementsVersion);
+    EXPECT_TRUE(CHIP_NO_ERROR == err);
```

Comment:
See my prior comment about changing `EXPECT_TRUE(CHIP_NO_ERROR == err)`

Reply:
also when comparing errors, `#include <lib/core/StringBuilderAdapters.h>`. It makes failures prettier (and unfortunately we have not found a central place to have this available to all tests by default)